lib/ethers/transaction.ex

```diff
@@ -257,22 +257,12 @@ defmodule Ethers.Transaction do
 
   defp do_post_process(_key, {:error, reason}), do: {:error, reason}
 
-  defp convert_to_binary(list) do
-    Enum.map(list, fn
-      "0x" <> _ = bin ->
-        bin
-        |> Utils.hex_decode!()
-        |> trim_leading()
-
-      l when is_list(l) ->
-        convert_to_binary(l)
-
-      nil ->
-        ""
+  defp hex_decode("0x" <> _ = bin) do
```

Comment:
I think it missed two other matching clauses here for the case of empty `input` - eg native token transfer.

```
 defp hex_decode(nil), do: ""

 defp hex_decode(item), do: item
```

Saw this error

```
** (FunctionClauseError) no function clause matching in Ethers.Transaction.hex_decode/1    
    
    The following arguments were given to Ethers.Transaction.hex_decode/1:
    
        # 1
        ""
    
    Attempted function clauses (showing 2 out of 2):
    
        defp hex_decode(<<"0x", _::binary>> = bin)
        defp hex_decode(list) when is_list(list)
    
    (ethers 0.3.0) lib/ethers/transaction.ex:264: Ethers.Transaction.hex_decode/1
    (ethers 0.3.0) lib/ethers/transaction.ex:211: Ethers.Transaction.to_rlp_list/1
    (ethers 0.3.0) lib/ethers/transaction.ex:106: Ethers.Transaction.encode/1
    (ethers 0.3.0) lib/ethers/signer/local.ex:36: Ethers.Signer.Local.sign_transaction/2
    (ethers 0.3.0) lib/ethers.ex:759: Ethers.use_signer/3
    (ethers 0.3.0) lib/ethers.ex:333: Ethers.send/2
    iex:1: (file)

```

Reply:
Fixed in d0b451c